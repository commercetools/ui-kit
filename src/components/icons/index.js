// This file is auto-generated using the 'generate-icon-exports.js' script
// so any changes made to this file manually will be lost the next time the
// script is executed
import withSizeProp from './with-size-prop';
import withThemeProp from './with-theme-prop';
import OrigAngleDownIcon from './svg/angle-down.react.svg';
import OrigAngleLeftIcon from './svg/angle-left.react.svg';
import OrigAngleRightIcon from './svg/angle-right.react.svg';
import OrigAngleThinLeftIcon from './svg/angle-thin-left.react.svg';
import OrigAngleThinRightIcon from './svg/angle-thin-right.react.svg';
import OrigAngleUpIcon from './svg/angle-up.react.svg';
import OrigArrowDownIcon from './svg/arrow-down.react.svg';
import OrigArrowLeftIcon from './svg/arrow-left.react.svg';
import OrigArrowLongDownIcon from './svg/arrow-long-down.react.svg';
import OrigArrowRightIcon from './svg/arrow-right.react.svg';
import OrigArrowTriangleDownIcon from './svg/arrow-triangle-down.react.svg';
import OrigArrowTriangleUpIcon from './svg/arrow-triangle-up.react.svg';
import OrigArrowUpIcon from './svg/arrow-up.react.svg';
import OrigArrowsIcon from './svg/arrows.react.svg';
import OrigBackIcon from './svg/back.react.svg';
import OrigBidirectionalArrowIcon from './svg/bidirectional-arrow.react.svg';
import OrigBinFilledIcon from './svg/bin-filled.react.svg';
import OrigBinLinearIcon from './svg/bin-linear.react.svg';
import OrigBoxIcon from './svg/box.react.svg';
import OrigBrainIcon from './svg/brain.react.svg';
import OrigCalendarIcon from './svg/calendar.react.svg';
import OrigCameraIcon from './svg/camera.react.svg';
import OrigCaretDownIcon from './svg/caret-down.react.svg';
import OrigCaretUpIcon from './svg/caret-up.react.svg';
import OrigCartIcon from './svg/cart.react.svg';
import OrigChainBrokenIcon from './svg/chain-broken.react.svg';
import OrigChainIcon from './svg/chain.react.svg';
import OrigCheckActiveIcon from './svg/check-active.react.svg';
import OrigCheckBoldIcon from './svg/check-bold.react.svg';
import OrigCheckInactiveIcon from './svg/check-inactive.react.svg';
import OrigCheckThinIcon from './svg/check-thin.react.svg';
import OrigCircleIcon from './svg/circle.react.svg';
import OrigClipboardIcon from './svg/clipboard.react.svg';
import OrigClockIcon from './svg/clock.react.svg';
import OrigCloseBoldIcon from './svg/close-bold.react.svg';
import OrigCloseIcon from './svg/close.react.svg';
import OrigCodeViewIcon from './svg/code-view.react.svg';
import OrigCoinsIcon from './svg/coins.react.svg';
import OrigColumnsIcon from './svg/columns.react.svg';
import OrigConnectedSquareIcon from './svg/connected-square.react.svg';
import OrigConnectedTriangleIcon from './svg/connected-triangle.react.svg';
import OrigCopyIcon from './svg/copy.react.svg';
import OrigCubeIcon from './svg/cube.react.svg';
import OrigCubesIcon from './svg/cubes.react.svg';
import OrigDotIcon from './svg/dot.react.svg';
import OrigDownloadIcon from './svg/download.react.svg';
import OrigDragDropIcon from './svg/drag-drop.react.svg';
import OrigDragIcon from './svg/drag.react.svg';
import OrigEditIcon from './svg/edit.react.svg';
import OrigErrorIcon from './svg/error.react.svg';
import OrigExpandIcon from './svg/expand.react.svg';
import OrigExportIcon from './svg/export.react.svg';
import OrigEyeCrossedIcon from './svg/eye-crossed.react.svg';
import OrigFilterIcon from './svg/filter.react.svg';
import OrigFlagFilledIcon from './svg/flag-filled.react.svg';
import OrigFlagLinearIcon from './svg/flag-linear.react.svg';
import OrigFlameIcon from './svg/flame.react.svg';
import OrigGearIcon from './svg/gear.react.svg';
import OrigGraphIcon from './svg/graph.react.svg';
import OrigGridIcon from './svg/grid.react.svg';
import OrigHeartIcon from './svg/heart.react.svg';
import OrigImportIcon from './svg/import.react.svg';
import OrigInfoIcon from './svg/info.react.svg';
import OrigInformationIcon from './svg/information.react.svg';
import OrigListIcon from './svg/list.react.svg';
import OrigLogoutIcon from './svg/logout.react.svg';
import OrigMailIcon from './svg/mail.react.svg';
import OrigMinimizeIcon from './svg/minimize.react.svg';
import OrigNestedViewIcon from './svg/nested-view.react.svg';
import OrigPagesIcon from './svg/pages.react.svg';
import OrigPaperBillInvertedIcon from './svg/paper-bill-inverted.react.svg';
import OrigPaperclipIcon from './svg/paperclip.react.svg';
import OrigPinFilledIcon from './svg/pin-filled.react.svg';
import OrigPinLinearIcon from './svg/pin-linear.react.svg';
import OrigPluginIcon from './svg/plugin.react.svg';
import OrigPlusBoldIcon from './svg/plus-bold.react.svg';
import OrigPlusThinIcon from './svg/plus-thin.react.svg';
import OrigRefreshIcon from './svg/refresh.react.svg';
import OrigRestoreIcon from './svg/restore.react.svg';
import OrigRevertIcon from './svg/revert.react.svg';
import OrigReviewIcon from './svg/review.react.svg';
import OrigRocketIcon from './svg/rocket.react.svg';
import OrigScreenGearIcon from './svg/screen-gear.react.svg';
import OrigScreenUserIcon from './svg/screen-user.react.svg';
import OrigSearchIcon from './svg/search.react.svg';
import OrigSortingIcon from './svg/sorting.react.svg';
import OrigSpeedometerIcon from './svg/speedometer.react.svg';
import OrigSplitIcon from './svg/split.react.svg';
import OrigStarIcon from './svg/star.react.svg';
import OrigSupportIcon from './svg/support.react.svg';
import OrigSwitcherIcon from './svg/switcher.react.svg';
import OrigTableIcon from './svg/table.react.svg';
import OrigTagMultiIcon from './svg/tag-multi.react.svg';
import OrigTagStackedIcon from './svg/tag-stacked.react.svg';
import OrigTagIcon from './svg/tag.react.svg';
import OrigTreeStructureIcon from './svg/tree-structure.react.svg';
import OrigTruckIcon from './svg/truck.react.svg';
import OrigUserFilledIcon from './svg/user-filled.react.svg';
import OrigUserLinearIcon from './svg/user-linear.react.svg';
import OrigVerifiedIcon from './svg/verified.react.svg';
import OrigWarningIcon from './svg/warning.react.svg';
import OrigWorldIcon from './svg/world.react.svg';

const AngleDownIcon = withThemeProp(withSizeProp(OrigAngleDownIcon));
const AngleLeftIcon = withThemeProp(withSizeProp(OrigAngleLeftIcon));
const AngleRightIcon = withThemeProp(withSizeProp(OrigAngleRightIcon));
const AngleThinLeftIcon = withThemeProp(withSizeProp(OrigAngleThinLeftIcon));
const AngleThinRightIcon = withThemeProp(withSizeProp(OrigAngleThinRightIcon));
const AngleUpIcon = withThemeProp(withSizeProp(OrigAngleUpIcon));
const ArrowDownIcon = withThemeProp(withSizeProp(OrigArrowDownIcon));
const ArrowLeftIcon = withThemeProp(withSizeProp(OrigArrowLeftIcon));
const ArrowLongDownIcon = withThemeProp(withSizeProp(OrigArrowLongDownIcon));
const ArrowRightIcon = withThemeProp(withSizeProp(OrigArrowRightIcon));
const ArrowTriangleDownIcon = withThemeProp(
  withSizeProp(OrigArrowTriangleDownIcon)
);
const ArrowTriangleUpIcon = withThemeProp(
  withSizeProp(OrigArrowTriangleUpIcon)
);
const ArrowUpIcon = withThemeProp(withSizeProp(OrigArrowUpIcon));
const ArrowsIcon = withThemeProp(withSizeProp(OrigArrowsIcon));
const BackIcon = withThemeProp(withSizeProp(OrigBackIcon));
const BidirectionalArrowIcon = withThemeProp(
  withSizeProp(OrigBidirectionalArrowIcon)
);
const BinFilledIcon = withThemeProp(withSizeProp(OrigBinFilledIcon));
const BinLinearIcon = withThemeProp(withSizeProp(OrigBinLinearIcon));
const BoxIcon = withThemeProp(withSizeProp(OrigBoxIcon));
const BrainIcon = withThemeProp(withSizeProp(OrigBrainIcon));
const CalendarIcon = withThemeProp(withSizeProp(OrigCalendarIcon));
const CameraIcon = withThemeProp(withSizeProp(OrigCameraIcon));
const CaretDownIcon = withThemeProp(withSizeProp(OrigCaretDownIcon));
const CaretUpIcon = withThemeProp(withSizeProp(OrigCaretUpIcon));
const CartIcon = withThemeProp(withSizeProp(OrigCartIcon));
const ChainBrokenIcon = withThemeProp(withSizeProp(OrigChainBrokenIcon));
const ChainIcon = withThemeProp(withSizeProp(OrigChainIcon));
const CheckActiveIcon = withThemeProp(withSizeProp(OrigCheckActiveIcon));
const CheckBoldIcon = withThemeProp(withSizeProp(OrigCheckBoldIcon));
const CheckInactiveIcon = withThemeProp(withSizeProp(OrigCheckInactiveIcon));
const CheckThinIcon = withThemeProp(withSizeProp(OrigCheckThinIcon));
const CircleIcon = withThemeProp(withSizeProp(OrigCircleIcon));
const ClipboardIcon = withThemeProp(withSizeProp(OrigClipboardIcon));
const ClockIcon = withThemeProp(withSizeProp(OrigClockIcon));
const CloseBoldIcon = withThemeProp(withSizeProp(OrigCloseBoldIcon));
const CloseIcon = withThemeProp(withSizeProp(OrigCloseIcon));
const CodeViewIcon = withThemeProp(withSizeProp(OrigCodeViewIcon));
const CoinsIcon = withThemeProp(withSizeProp(OrigCoinsIcon));
const ColumnsIcon = withThemeProp(withSizeProp(OrigColumnsIcon));
const ConnectedSquareIcon = withThemeProp(
  withSizeProp(OrigConnectedSquareIcon)
);
const ConnectedTriangleIcon = withThemeProp(
  withSizeProp(OrigConnectedTriangleIcon)
);
const CopyIcon = withThemeProp(withSizeProp(OrigCopyIcon));
const CubeIcon = withThemeProp(withSizeProp(OrigCubeIcon));
const CubesIcon = withThemeProp(withSizeProp(OrigCubesIcon));
const DotIcon = withThemeProp(withSizeProp(OrigDotIcon));
const DownloadIcon = withThemeProp(withSizeProp(OrigDownloadIcon));
const DragDropIcon = withThemeProp(withSizeProp(OrigDragDropIcon));
const DragIcon = withThemeProp(withSizeProp(OrigDragIcon));
const EditIcon = withThemeProp(withSizeProp(OrigEditIcon));
const ErrorIcon = withThemeProp(withSizeProp(OrigErrorIcon));
const ExpandIcon = withThemeProp(withSizeProp(OrigExpandIcon));
const ExportIcon = withThemeProp(withSizeProp(OrigExportIcon));
const EyeCrossedIcon = withThemeProp(withSizeProp(OrigEyeCrossedIcon));
const FilterIcon = withThemeProp(withSizeProp(OrigFilterIcon));
const FlagFilledIcon = withThemeProp(withSizeProp(OrigFlagFilledIcon));
const FlagLinearIcon = withThemeProp(withSizeProp(OrigFlagLinearIcon));
const FlameIcon = withThemeProp(withSizeProp(OrigFlameIcon));
const GearIcon = withThemeProp(withSizeProp(OrigGearIcon));
const GraphIcon = withThemeProp(withSizeProp(OrigGraphIcon));
const GridIcon = withThemeProp(withSizeProp(OrigGridIcon));
const HeartIcon = withThemeProp(withSizeProp(OrigHeartIcon));
const ImportIcon = withThemeProp(withSizeProp(OrigImportIcon));
const InfoIcon = withThemeProp(withSizeProp(OrigInfoIcon));
const InformationIcon = withThemeProp(withSizeProp(OrigInformationIcon));
const ListIcon = withThemeProp(withSizeProp(OrigListIcon));
const LogoutIcon = withThemeProp(withSizeProp(OrigLogoutIcon));
const MailIcon = withThemeProp(withSizeProp(OrigMailIcon));
const MinimizeIcon = withThemeProp(withSizeProp(OrigMinimizeIcon));
const NestedViewIcon = withThemeProp(withSizeProp(OrigNestedViewIcon));
const PagesIcon = withThemeProp(withSizeProp(OrigPagesIcon));
const PaperBillInvertedIcon = withThemeProp(
  withSizeProp(OrigPaperBillInvertedIcon)
);
const PaperclipIcon = withThemeProp(withSizeProp(OrigPaperclipIcon));
const PinFilledIcon = withThemeProp(withSizeProp(OrigPinFilledIcon));
const PinLinearIcon = withThemeProp(withSizeProp(OrigPinLinearIcon));
const PluginIcon = withThemeProp(withSizeProp(OrigPluginIcon));
const PlusBoldIcon = withThemeProp(withSizeProp(OrigPlusBoldIcon));
const PlusThinIcon = withThemeProp(withSizeProp(OrigPlusThinIcon));
const RefreshIcon = withThemeProp(withSizeProp(OrigRefreshIcon));
const RestoreIcon = withThemeProp(withSizeProp(OrigRestoreIcon));
const RevertIcon = withThemeProp(withSizeProp(OrigRevertIcon));
const ReviewIcon = withThemeProp(withSizeProp(OrigReviewIcon));
const RocketIcon = withThemeProp(withSizeProp(OrigRocketIcon));
const ScreenGearIcon = withThemeProp(withSizeProp(OrigScreenGearIcon));
const ScreenUserIcon = withThemeProp(withSizeProp(OrigScreenUserIcon));
const SearchIcon = withThemeProp(withSizeProp(OrigSearchIcon));
const SortingIcon = withThemeProp(withSizeProp(OrigSortingIcon));
const SpeedometerIcon = withThemeProp(withSizeProp(OrigSpeedometerIcon));
const SplitIcon = withThemeProp(withSizeProp(OrigSplitIcon));
const StarIcon = withThemeProp(withSizeProp(OrigStarIcon));
const SupportIcon = withThemeProp(withSizeProp(OrigSupportIcon));
const SwitcherIcon = withThemeProp(withSizeProp(OrigSwitcherIcon));
const TableIcon = withThemeProp(withSizeProp(OrigTableIcon));
const TagMultiIcon = withThemeProp(withSizeProp(OrigTagMultiIcon));
const TagStackedIcon = withThemeProp(withSizeProp(OrigTagStackedIcon));
const TagIcon = withThemeProp(withSizeProp(OrigTagIcon));
const TreeStructureIcon = withThemeProp(withSizeProp(OrigTreeStructureIcon));
const TruckIcon = withThemeProp(withSizeProp(OrigTruckIcon));
const UserFilledIcon = withThemeProp(withSizeProp(OrigUserFilledIcon));
const UserLinearIcon = withThemeProp(withSizeProp(OrigUserLinearIcon));
const VerifiedIcon = withThemeProp(withSizeProp(OrigVerifiedIcon));
const WarningIcon = withThemeProp(withSizeProp(OrigWarningIcon));
const WorldIcon = withThemeProp(withSizeProp(OrigWorldIcon));

AngleDownIcon.displayName = 'AngleDownIcon';
AngleLeftIcon.displayName = 'AngleLeftIcon';
AngleRightIcon.displayName = 'AngleRightIcon';
AngleThinLeftIcon.displayName = 'AngleThinLeftIcon';
AngleThinRightIcon.displayName = 'AngleThinRightIcon';
AngleUpIcon.displayName = 'AngleUpIcon';
ArrowDownIcon.displayName = 'ArrowDownIcon';
ArrowLeftIcon.displayName = 'ArrowLeftIcon';
ArrowLongDownIcon.displayName = 'ArrowLongDownIcon';
ArrowRightIcon.displayName = 'ArrowRightIcon';
ArrowTriangleDownIcon.displayName = 'ArrowTriangleDownIcon';
ArrowTriangleUpIcon.displayName = 'ArrowTriangleUpIcon';
ArrowUpIcon.displayName = 'ArrowUpIcon';
ArrowsIcon.displayName = 'ArrowsIcon';
BackIcon.displayName = 'BackIcon';
BidirectionalArrowIcon.displayName = 'BidirectionalArrowIcon';
BinFilledIcon.displayName = 'BinFilledIcon';
BinLinearIcon.displayName = 'BinLinearIcon';
BoxIcon.displayName = 'BoxIcon';
BrainIcon.displayName = 'BrainIcon';
CalendarIcon.displayName = 'CalendarIcon';
CameraIcon.displayName = 'CameraIcon';
CaretDownIcon.displayName = 'CaretDownIcon';
CaretUpIcon.displayName = 'CaretUpIcon';
CartIcon.displayName = 'CartIcon';
ChainBrokenIcon.displayName = 'ChainBrokenIcon';
ChainIcon.displayName = 'ChainIcon';
CheckActiveIcon.displayName = 'CheckActiveIcon';
CheckBoldIcon.displayName = 'CheckBoldIcon';
CheckInactiveIcon.displayName = 'CheckInactiveIcon';
CheckThinIcon.displayName = 'CheckThinIcon';
CircleIcon.displayName = 'CircleIcon';
ClipboardIcon.displayName = 'ClipboardIcon';
ClockIcon.displayName = 'ClockIcon';
CloseBoldIcon.displayName = 'CloseBoldIcon';
CloseIcon.displayName = 'CloseIcon';
CodeViewIcon.displayName = 'CodeViewIcon';
CoinsIcon.displayName = 'CoinsIcon';
ColumnsIcon.displayName = 'ColumnsIcon';
ConnectedSquareIcon.displayName = 'ConnectedSquareIcon';
ConnectedTriangleIcon.displayName = 'ConnectedTriangleIcon';
CopyIcon.displayName = 'CopyIcon';
CubeIcon.displayName = 'CubeIcon';
CubesIcon.displayName = 'CubesIcon';
DotIcon.displayName = 'DotIcon';
DownloadIcon.displayName = 'DownloadIcon';
DragDropIcon.displayName = 'DragDropIcon';
DragIcon.displayName = 'DragIcon';
EditIcon.displayName = 'EditIcon';
ErrorIcon.displayName = 'ErrorIcon';
ExpandIcon.displayName = 'ExpandIcon';
ExportIcon.displayName = 'ExportIcon';
EyeCrossedIcon.displayName = 'EyeCrossedIcon';
FilterIcon.displayName = 'FilterIcon';
FlagFilledIcon.displayName = 'FlagFilledIcon';
FlagLinearIcon.displayName = 'FlagLinearIcon';
FlameIcon.displayName = 'FlameIcon';
GearIcon.displayName = 'GearIcon';
GraphIcon.displayName = 'GraphIcon';
GridIcon.displayName = 'GridIcon';
HeartIcon.displayName = 'HeartIcon';
ImportIcon.displayName = 'ImportIcon';
InfoIcon.displayName = 'InfoIcon';
InformationIcon.displayName = 'InformationIcon';
ListIcon.displayName = 'ListIcon';
LogoutIcon.displayName = 'LogoutIcon';
MailIcon.displayName = 'MailIcon';
MinimizeIcon.displayName = 'MinimizeIcon';
NestedViewIcon.displayName = 'NestedViewIcon';
PagesIcon.displayName = 'PagesIcon';
PaperBillInvertedIcon.displayName = 'PaperBillInvertedIcon';
PaperclipIcon.displayName = 'PaperclipIcon';
PinFilledIcon.displayName = 'PinFilledIcon';
PinLinearIcon.displayName = 'PinLinearIcon';
PluginIcon.displayName = 'PluginIcon';
PlusBoldIcon.displayName = 'PlusBoldIcon';
PlusThinIcon.displayName = 'PlusThinIcon';
RefreshIcon.displayName = 'RefreshIcon';
RestoreIcon.displayName = 'RestoreIcon';
RevertIcon.displayName = 'RevertIcon';
ReviewIcon.displayName = 'ReviewIcon';
RocketIcon.displayName = 'RocketIcon';
ScreenGearIcon.displayName = 'ScreenGearIcon';
ScreenUserIcon.displayName = 'ScreenUserIcon';
SearchIcon.displayName = 'SearchIcon';
SortingIcon.displayName = 'SortingIcon';
SpeedometerIcon.displayName = 'SpeedometerIcon';
SplitIcon.displayName = 'SplitIcon';
StarIcon.displayName = 'StarIcon';
SupportIcon.displayName = 'SupportIcon';
SwitcherIcon.displayName = 'SwitcherIcon';
TableIcon.displayName = 'TableIcon';
TagMultiIcon.displayName = 'TagMultiIcon';
TagStackedIcon.displayName = 'TagStackedIcon';
TagIcon.displayName = 'TagIcon';
TreeStructureIcon.displayName = 'TreeStructureIcon';
TruckIcon.displayName = 'TruckIcon';
UserFilledIcon.displayName = 'UserFilledIcon';
UserLinearIcon.displayName = 'UserLinearIcon';
VerifiedIcon.displayName = 'VerifiedIcon';
WarningIcon.displayName = 'WarningIcon';
WorldIcon.displayName = 'WorldIcon';

export { AngleDownIcon };
export { AngleLeftIcon };
export { AngleRightIcon };
export { AngleThinLeftIcon };
export { AngleThinRightIcon };
export { AngleUpIcon };
export { ArrowDownIcon };
export { ArrowLeftIcon };
export { ArrowLongDownIcon };
export { ArrowRightIcon };
export { ArrowTriangleDownIcon };
export { ArrowTriangleUpIcon };
export { ArrowUpIcon };
export { ArrowsIcon };
export { BackIcon };
export { BidirectionalArrowIcon };
export { BinFilledIcon };
export { BinLinearIcon };
export { BoxIcon };
export { BrainIcon };
export { CalendarIcon };
export { CameraIcon };
export { CaretDownIcon };
export { CaretUpIcon };
export { CartIcon };
export { ChainBrokenIcon };
export { ChainIcon };
export { CheckActiveIcon };
export { CheckBoldIcon };
export { CheckInactiveIcon };
export { CheckThinIcon };
export { CircleIcon };
export { ClipboardIcon };
export { ClockIcon };
export { CloseBoldIcon };
export { CloseIcon };
export { CodeViewIcon };
export { CoinsIcon };
export { ColumnsIcon };
export { ConnectedSquareIcon };
export { ConnectedTriangleIcon };
export { CopyIcon };
export { CubeIcon };
export { CubesIcon };
export { DotIcon };
export { DownloadIcon };
export { DragDropIcon };
export { DragIcon };
export { EditIcon };
export { ErrorIcon };
export { ExpandIcon };
export { ExportIcon };
export { EyeCrossedIcon };
export { FilterIcon };
export { FlagFilledIcon };
export { FlagLinearIcon };
export { FlameIcon };
export { GearIcon };
export { GraphIcon };
export { GridIcon };
export { HeartIcon };
export { ImportIcon };
export { InfoIcon };
export { InformationIcon };
export { ListIcon };
export { LogoutIcon };
export { MailIcon };
export { MinimizeIcon };
export { NestedViewIcon };
export { PagesIcon };
export { PaperBillInvertedIcon };
export { PaperclipIcon };
export { PinFilledIcon };
export { PinLinearIcon };
export { PluginIcon };
export { PlusBoldIcon };
export { PlusThinIcon };
export { RefreshIcon };
export { RestoreIcon };
export { RevertIcon };
export { ReviewIcon };
export { RocketIcon };
export { ScreenGearIcon };
export { ScreenUserIcon };
export { SearchIcon };
export { SortingIcon };
export { SpeedometerIcon };
export { SplitIcon };
export { StarIcon };
export { SupportIcon };
export { SwitcherIcon };
export { TableIcon };
export { TagMultiIcon };
export { TagStackedIcon };
export { TagIcon };
export { TreeStructureIcon };
export { TruckIcon };
export { UserFilledIcon };
export { UserLinearIcon };
export { VerifiedIcon };
export { WarningIcon };
export { WorldIcon };
