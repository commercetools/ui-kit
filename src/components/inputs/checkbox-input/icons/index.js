// NOTE: explicit names for component names
import CheckboxDefaultIcon from './checkbox-default.react.svg';
import CheckboxCheckedIcon from './checkbox-checked.react.svg';
import CheckboxIndeterminateIcon from './checkbox-indeterminate.react.svg';

export default {
  Unchecked: CheckboxDefaultIcon,
  Checked: CheckboxCheckedIcon,
  Indeterminate: CheckboxIndeterminateIcon,
};
