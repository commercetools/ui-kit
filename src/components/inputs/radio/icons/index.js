// NOTE: explicit names for component names
import RadioOptionDefaultIcon from './radio-option-default.react.svg';
import RadioOptionCheckedIcon from './radio-option-checked.react.svg';

export default {
  Default: RadioOptionDefaultIcon,
  Checked: RadioOptionCheckedIcon,
};
