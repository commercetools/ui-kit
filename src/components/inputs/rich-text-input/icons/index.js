import createStyledIcon from '../../../icons/create-styled-icon';
import OrigBoldIcon from './svg/bold.react.svg';
import OrigItalicIcon from './svg/italic.react.svg';
import OrigOrderedListIcon from './svg/ordered-list.react.svg';
import OrigMoreStylesIcon from './svg/more-styles.react.svg';
import OrigUnderlineIcon from './svg/underline.react.svg';
import OrigUnorderedListIcon from './svg/unordered-list.react.svg';
import OrigStrikethroughIcon from './svg/strikethrough.react.svg';
import OrigSuperscriptIcon from './svg/superscript.react.svg';
import OrigSubscriptIcon from './svg/subscript.react.svg';
import OrigRedoIcon from './svg/redo.react.svg';

export const BoldIcon = createStyledIcon(OrigBoldIcon, 'BoldIcon');
export const ItalicIcon = createStyledIcon(OrigItalicIcon, 'ItalicIcon');
export const MoreStylesIcon = createStyledIcon(
  OrigMoreStylesIcon,
  'MoreStylesIcon'
);
export const OrderedListIcon = createStyledIcon(
  OrigOrderedListIcon,
  'OrderedListIcon'
);
export const UnderlineIcon = createStyledIcon(
  OrigUnderlineIcon,
  'UnderlineIcon'
);
export const UnorderedListIcon = createStyledIcon(
  OrigUnorderedListIcon,
  'UnorderedListIcon'
);

export const StrikethroughIcon = createStyledIcon(
  OrigStrikethroughIcon,
  'StrikethroughIcon'
);
export const SuperscriptIcon = createStyledIcon(
  OrigSuperscriptIcon,
  'SuperscriptIcon'
);
export const SubscriptIcon = createStyledIcon(
  OrigSubscriptIcon,
  'SubscriptIcon'
);

export const RedoIcon = createStyledIcon(OrigRedoIcon, 'RedoIcon');
