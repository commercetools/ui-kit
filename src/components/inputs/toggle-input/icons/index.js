import ToggleButtonIcon from './toggle-button-default.react.svg';
import ToggleBarIcon from './toggle-bar-default.react.svg';

export default {
  ButtonDefault: ToggleButtonIcon,
  BarDefault: ToggleBarIcon,
};
