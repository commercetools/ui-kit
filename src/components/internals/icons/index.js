// This file is auto-generated using the 'generate-icon-exports.js' script
// so any changes made to this file manually will be lost the next time the
// script is executed
export { default as BoldIcon } from './components/bold';
export { default as CheckedIcon } from './components/checked';
export { default as ExpandIcon } from './components/expand';
export { default as IndeterminateIcon } from './components/indeterminate';
export { default as ItalicIcon } from './components/italic';
export { default as MoreStylesIcon } from './components/more-styles';
export { default as OrderedListIcon } from './components/ordered-list';
export { default as RadioOptionCheckedIcon } from './components/radio-option-checked';
export { default as RadioOptionUncheckedIcon } from './components/radio-option-unchecked';
export { default as RedoIcon } from './components/redo';
export { default as StrikethroughIcon } from './components/strikethrough';
export { default as SubscriptIcon } from './components/subscript';
export { default as SuperscriptIcon } from './components/superscript';
export { default as UncheckedIcon } from './components/unchecked';
export { default as UnderlineIcon } from './components/underline';
export { default as UndoIcon } from './components/undo';
export { default as UnorderedListIcon } from './components/unordered-list';
