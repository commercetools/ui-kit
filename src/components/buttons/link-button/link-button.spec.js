import React from 'react';
import { shallow } from 'enzyme';
import Text from '../../typography/text';
import { PlusThinIcon } from '../../icons';
import LinkButton from './link-button';

const createTestProps = props => ({
  label: 'Accessibility text',
  to: '/foo/bar',
  iconLeft: <PlusThinIcon size="medium" />,
  isDisabled: false,
  ...props,
});

const createEvent = props => ({
  preventDefault: jest.fn(),
  ...props,
});

describe('rendering', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<LinkButton {...props} />);
  });

  it('outputs correct tree', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <Link>', () => {
    expect(wrapper).toRender('Link');
  });

  describe('onClick', () => {
    const testEvent = createEvent();
    beforeEach(() => {
      wrapper.find('Link').simulate('click', testEvent);
    });
    it('should have undefined onClick prop', () => {
      expect(wrapper.find('Link')).toHaveProp('onClick', undefined);
    });
    it('when `onClick` is fired, event default is not prevented', () => {
      expect(testEvent.preventDefault).not.toHaveBeenCalled();
    });
  });

  it('should pass the url to <Link>', () => {
    expect(wrapper.find('Link')).toHaveProp('to', '/foo/bar');
  });

  it('should render a <Text.Body>', () => {
    expect(
      wrapper.containsMatchingElement(
        <Text.Body>{'Accessibility text'}</Text.Body>
      )
    ).toBe(true);
  });

  it('should render the icon', () => {
    expect(wrapper).toRender(PlusThinIcon);
  });

  describe('with data-* props', () => {
    beforeEach(() => {
      props = createTestProps({
        'data-track-component': 'LinkButton',
        'data-track-label': 'LinkButton',
        'data-track-event': 'click',
        'data-test': 'link-button',
      });
      wrapper = shallow(<LinkButton {...props} />);
    });
    it('should apply given `data-track-component` to LinkButton', () => {
      expect(wrapper.find('Link')).toHaveProp(
        'data-track-component',
        expect.stringMatching('LinkButton')
      );
    });

    it('should apply given `data-track-event` to AccessibleButton', () => {
      expect(wrapper.find('Link')).toHaveProp(
        'data-track-event',
        expect.stringMatching('click')
      );
    });
    it('should apply given `data-track-label` to AccessibleButton', () => {
      expect(wrapper.find('Link')).toHaveProp(
        'data-track-label',
        expect.stringMatching('LinkButton')
      );
    });
    it('should apply given `data-test` to AccessibleButton', () => {
      expect(wrapper.find('Link')).toHaveProp(
        'data-test',
        expect.stringMatching('link-button')
      );
    });
  });

  describe('without data-* props', () => {
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<LinkButton {...props} />);
    });
    it('should apply default `data-track-component` to LinkButton', () => {
      expect(wrapper.find('Link')).toHaveProp(
        'data-track-component',
        expect.stringMatching('LinkButton')
      );
    });
  });

  describe('if disabled', () => {
    beforeEach(() => {
      props = createTestProps({ isDisabled: true });
      wrapper = shallow(<LinkButton {...props} />);
    });
    it('renders a <Link> with disabled class', () => {
      expect(wrapper.find('Link')).toHaveClassName('disabled');
    });

    describe('onClick', () => {
      const testEvent = createEvent();
      beforeEach(() => {
        wrapper.find('Link').simulate('click', testEvent);
      });
      it('renders with a onClick prop', () => {
        expect(wrapper.find('Link')).toHaveProp('onClick');
      });
      it('when `onClick` is fired, event default is prevented', () => {
        expect(testEvent.preventDefault).toHaveBeenCalled();
      });
    });

    it('renders the icon with "grey" theme', () => {
      expect(wrapper.find(PlusThinIcon)).toHaveProp('theme', 'grey');
    });
  });
});
