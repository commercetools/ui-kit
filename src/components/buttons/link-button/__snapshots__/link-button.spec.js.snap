// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering outputs correct tree 1`] = `
<Link
  className="button"
  data-track-component="LinkButton"
  replace={false}
  to="/foo/bar"
>
  <PlusThinIcon
    size="medium"
    theme="green"
  />
  <TextBody>
    Accessibility text
  </TextBody>
</Link>
`;
